viscy/translation/evaluation.py

```diff
@@ -12,43 +15,254 @@
+        if "infection_condition" in batch:
+            metrics_dict["infection_condition"] = batch["infection_condition"]
+
+        self.logger.log_metrics(metrics_dict)
+
+    def _compute_3d_metrics(self, pred, target, pred_binary, target_binary, batch):
+        """Compute and log metrics for 3D segmentation."""
+        unique_instances_target = np.unique(target)
+        unique_instances_pred = np.unique(pred)
+
+        _logger.debug(
+            f"Unique instances: {unique_instances_target} and {unique_instances_pred}"
         )
+
+        ## Measuring Panoptic Quality
+        evaluator = Panoptica_Evaluator(
+            expected_input=InputType.UnmatchedInstancePair,
+            instance_metrics=[Metric.DSC, Metric.IoU],
+            decision_metric=Metric.DSC,
+            decision_threshold=0.5,
+            log_times=True,
+        )
+        result = evaluator.evaluate(pred, target, verbose=False)
+        result = result.to_dict()
+        _logger.debug(result)
+
+        # Create metrics dictionary
+        metrics_dict = {
+            "position": batch["position_idx"][0],
+            "time": batch["time_idx"][0],
+            "target_instances": unique_instances_target,
+            "pred_instances": unique_instances_pred,
+            **result,
+        }
+
+        # Add metadata if available
+        if "cell_type" in batch:
+            metrics_dict["cell_type"] = batch["cell_type"]
+        if "organelle" in batch:
+            metrics_dict["organelle"] = batch["organelle"]
+        if "infection_condition" in batch:
+            metrics_dict["infection_condition"] = batch["infection_condition"]
+
+        self.logger.log_metrics(metrics_dict)
+
+
+class BiologicalMetrics(LightningModule):
+    """Test runner for biological metrics."""
+
+    def __init__(self, aggregate_epoch: bool = False) -> None:
+        super().__init__()
+        self.aggregate_epoch = aggregate_epoch
+
+    def test_step(self, batch: SegmentationSample, batch_idx: int) -> None:
+        # TODO: Implement biological metrics (i.e regionprops logic)
+        raise NotImplementedError("Biological metrics not implemented")
+
+
+class IntensityMetrics(LightningModule):
+    """Test runner for intensity metrics.
+
+    Parameters
+    ----------
+    metrics : list[str], optional
+        List of metrics to compute, by default ["mae", "mse", "ssim", "pearson"]
+    aggregate_epoch : bool, optional
+        Whether to aggregate results over the entire epoch, by default False
+    """
+
+    def __init__(
+        self,
+        metrics: list[str] = ["mae", "mse", "ssim", "pearson"],
+        aggregate_epoch: bool = False,
+    ) -> None:
+        super().__init__()
+        self.metrics = metrics
+        self.aggregate_epoch = aggregate_epoch
+        self._validate_metrics()
+
+    def _validate_metrics(self):
+        """Validate the metrics parameter."""
+        valid_metrics = ["mae", "mse", "ssim", "ms_ssim", "pearson", "cosine"]
```

Comment:
These should be documented in the signature of `__init__`, i.e. `metrics: list[Literal["mae", "mse", "ssim", "ms_ssim", "pearson", "cosine"]]`.